### DISCUSSION TITLE
There are cases when - either a table or a view (especially full outer joined tables which create a view) allow for all columns to be null, with no unique key - such cases should be allowed.  A workaround would have been to define several models/views in prisma, which actually point to the same db object (view or table), however, even this is not possible...

### DISCUSSION THREAD

**Discussion by lukzard** [prisma/prisma#23732] [Q&A]
There are cases when - either a table or a view (especially full outer joined tables which create a view) allow for all columns to be null, with no unique key - such cases should be allowed.

A workaround would have been to define several models/views in prisma, which actually point to the same db object (view or table), however, even this is not possible...

**Comment by ludralph** [comment]
Hi @lukzard 👋 

Thank you for raising this question.

> There are cases when - either a table or a view (especially full outer joined tables which create a view) allow for all columns to be null, with no unique key - such cases should be allowed.

You can create a [view](https://www.prisma.io/docs/orm/prisma-schema/data-model/views) in the underlying database that allows for all columns to be null, with no unique key. You would need to use introspection to generate view blocks in your Prisma Schema to represent those views. See this section of the [documentation](https://www.prisma.io/docs/orm/prisma-schema/data-model/views#use-introspection). Do note that to be able to use the introspected view in Prisma Client, you will need to select and define one or multiple of the fields as the unique identifier.

m

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
